### ISSUE TITLE
1.20

### ISSUE THREAD

**Issue by ZConnorZ** [Sashie/skript-yaml#46]
Does not work on 1.20.

**Comment by memosoff** [comment]
🥺🥺🥺

**Comment by ShaneBeee** [comment]
to add context, here is the console error
```
[09:51:25 WARN]: java.lang.NoSuchMethodException: org.yaml.snakeyaml.representer.BaseRepresenter.representMapping(org.yaml.snakeyaml.nodes.Tag,java.util.Map,java.lang.Boolean)
[09:51:25 WARN]: 	at java.base/java.lang.Class.getDeclaredMethod(Class.java:2728)
[09:51:25 WARN]: 	at skript-yaml-1.4.1.jar//me.sashie.skriptyaml.utils.yaml.SkriptYamlRepresenter.<clinit>(SkriptYamlRepresenter.java:35)
[09:51:25 WARN]: 	at skript-yaml-1.4.1.jar//me.sashie.skriptyaml.SkriptYaml.onEnable(SkriptYaml.java:137)
[09:51:25 WARN]: 	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:281)
[09:51:25 WARN]: 	at io.papermc.paper.plugin.manager.PaperPluginInstanceManager.enablePlugin(PaperPluginInstanceManager.java:189)
[09:51:25 WARN]: 	at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.enablePlugin(PaperPluginManagerImpl.java:104)
[09:51:25 WARN]: 	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:507)
[09:51:25 WARN]: 	at org.bukkit.craftbukkit.v1_20_R1.CraftServer.enablePlugin(CraftServer.java:563)
[09:51:25 WARN]: 	at org.bukkit.craftbukkit.v1_20_R1.CraftServer.enablePlugins(CraftServer.java:474)
[09:51:25 WARN]: 	at org.bukkit.craftbukkit.v1_20_R1.CraftServer.reload(CraftServer.java:1039)
[09:51:25 WARN]: 	at org.bukkit.Bukkit.reload(Bukkit.java:952)
[09:51:25 WARN]: 	at org.bukkit.command.defaults.ReloadCommand.execute(ReloadCommand.java:54)
[09:51:25 WARN]: 	at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:155)
[09:51:25 WARN]: 	at org.bukkit.craftbukkit.v1_20_R1.CraftServer.dispatchCommand(CraftServer.java:929)
[09:51:25 WARN]: 	at org.bukkit.craftbukkit.v1_20_R1.CraftServer.dispatchServerCommand(CraftServer.java:892)
[09:51:25 WARN]: 	at net.minecraft.server.dedicated.DedicatedServer.bf(DedicatedServer.java:501)
[09:51:25 WARN]: 	at net.minecraft.server.dedicated.DedicatedServer.b(DedicatedServer.java:448)
[09:51:25 WARN]: 	at net.minecraft.server.MinecraftServer.a(MinecraftServer.java:1396)
[09:51:25 WARN]: 	at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1173)
[09:51:25 WARN]: 	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:320)
[09:51:25 WARN]: 	at java.base/java.lang.Thread.run(Thread.java:1589)
[09:51:25 ERROR]: Error occurred while enabling skript-yaml v1.4.1 (Is it up to date?)
java.lang.NoSuchMethodError: org.yaml.snakeyaml.representer.Representer: method 'void <init>()' not found
	at me.sashie.skriptyaml.utils.yaml.SkriptYamlRepresenter.<init>(SkriptYamlRepresenter.java:47) ~[skript-yaml-1.4.1.jar:?]
	at me.sashie.skriptyaml.SkriptYaml.onEnable(SkriptYaml.java:137) ~[skript-yaml-1.4.1.jar:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:281) ~[paper-api-1.20-R0.1-SNAPSHOT.jar:?]
	at io.papermc.paper.plugin.manager.PaperPluginInstanceManager.enablePlugin(PaperPluginInstanceManager.java:189) ~[paper-1.20.jar:git-Paper-5]
	at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.enablePlugin(PaperPluginManagerImpl.java:104) ~[paper-1.20.jar:git-Paper-5]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:507) ~[paper-api-1.20-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.craftbukkit.v1_20_R1.CraftServer.enablePlugin(CraftServer.java:563) ~[paper-1.20.jar:git-Paper-5]
	at org.bukkit.craftbukkit.v1_20_R1.CraftServer.enablePlugins(CraftServer.java:474) ~[paper-1.20.jar:git-Paper-5]
	at org.bukkit.craftbukkit.v1_20_R1.CraftServer.reload(CraftServer.java:1039) ~[paper-1.20.jar:git-Paper-5]
	at org.bukkit.Bukkit.reload(Bukkit.java:952) ~[paper-api-1.20-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.command.defaults.ReloadCommand.execute(ReloadCommand.java:54) ~[paper-api-1.20-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:155) ~[paper-api-1.20-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.craftbukkit.v1_20_R1.CraftServer.dispatchCommand(CraftServer.java:929) ~[paper-1.20.jar:git-Paper-5]
	at org.bukkit.craftbukkit.v1_20_R1.CraftServer.dispatchServerCommand(CraftServer.java:892) ~[paper-1.20.jar:git-Paper-5]
	at net.minecraft.server.dedicated.DedicatedServer.handleConsoleInputs(DedicatedServer.java:501) ~[paper-1.20.jar:git-Paper-5]
	at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:448) ~[paper-1.20.jar:git-Paper-5]
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1396) ~[paper-1.20.jar:git-Paper-5]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1173) ~[paper-1.20.jar:git-Paper-5]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:320) ~[paper-1.20.jar:git-Paper-5]
	at java.lang.Thread.run(Thread.java:1589) ~[?:?]
```
I believe Bukkit updated SnakeYaml recently

**Comment by memosoff** [comment]
> to add context, here is the console error
> 
> ```
> [09:51:25 WARN]: java.lang.NoSuchMethodException: org.yaml.snakeyaml.representer.BaseRepresenter.representMapping(org.yaml.snakeyaml.nodes.Tag,java.util.Map,java.lang.Boolean)
> [09:51:25 WARN]: 	at java.base/java.lang.Class.getDeclaredMethod(Class.java:2728)
> [09:51:25 WARN]: 	at skript-yaml-1.4.1.jar//me.sashie.skriptyaml.utils.yaml.SkriptYamlRepresenter.<clinit>(SkriptYamlRepresenter.java:35)
> [09:51:25 WARN]: 	at skript-yaml-1.4.1.jar//me.sashie.skriptyaml.SkriptYaml.onEnable(SkriptYaml.java:137)
> [09:51:25 WARN]: 	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:281)
> [09:51:25 WARN]: 	at io.papermc.paper.plugin.manager.PaperPluginInstanceManager.enablePlugin(PaperPluginInstanceManager.java:189)
> [09:51:25 WARN]: 	at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.enablePlugin(PaperPluginManagerImpl.java:104)
> [09:51:25 WARN]: 	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:507)
> [09:51:25 WARN]: 	at org.bukkit.craftbukkit.v1_20_R1.CraftServer.enablePlugin(CraftServer.java:563)
> [09:51:25 WARN]: 	at org.bukkit.craftbukkit.v1_20_R1.CraftServer.enablePlugins(CraftServer.java:474)
> [09:51:25 WARN]: 	at org.bukkit.craftbukkit.v1_20_R1.CraftServer.reload(CraftServer.java:1039)
> [09:51:25 WARN]: 	at org.bukkit.Bukkit.reload(Bukkit.java:952)
> [09:51:25 WARN]: 	at org.bukkit.command.defaults.ReloadCommand.execute(ReloadCommand.java:54)
> [09:51:25 WARN]: 	at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:155)
> [09:51:25 WARN]: 	at org.bukkit.craftbukkit.v1_20_R1.CraftServer.dispatchCommand(CraftServer.java:929)
> [09:51:25 WARN]: 	at org.bukkit.craftbukkit.v1_20_R1.CraftServer.dispatchServerCommand(CraftServer.java:892)
> [09:51:25 WARN]: 	at net.minecraft.server.dedicated.DedicatedServer.bf(DedicatedServer.java:501)
> [09:51:25 WARN]: 	at net.minecraft.server.dedicated.DedicatedServer.b(DedicatedServer.java:448)
> [09:51:25 WARN]: 	at net.minecraft.server.MinecraftServer.a(MinecraftServer.java:1396)
> [09:51:25 WARN]: 	at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1173)
> [09:51:25 WARN]: 	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:320)
> [09:51:25 WARN]: 	at java.base/java.lang.Thread.run(Thread.java:1589)
> [09:51:25 ERROR]: Error occurred while enabling skript-yaml v1.4.1 (Is it up to date?)
> java.lang.NoSuchMethodError: org.yaml.snakeyaml.representer.Representer: method 'void <init>()' not found
> 	at me.sashie.skriptyaml.utils.yaml.SkriptYamlRepresenter.<init>(SkriptYamlRepresenter.java:47) ~[skript-yaml-1.4.1.jar:?]
> 	at me.sashie.skriptyaml.SkriptYaml.onEnable(SkriptYaml.java:137) ~[skript-yaml-1.4.1.jar:?]
> 	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:281) ~[paper-api-1.20-R0.1-SNAPSHOT.jar:?]
> 	at io.papermc.paper.plugin.manager.PaperPluginInstanceManager.enablePlugin(PaperPluginInstanceManager.java:189) ~[paper-1.20.jar:git-Paper-5]
> 	at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.enablePlugin(PaperPluginManagerImpl.java:104) ~[paper-1.20.jar:git-Paper-5]
> 	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:507) ~[paper-api-1.20-R0.1-SNAPSHOT.jar:?]
> 	at org.bukkit.craftbukkit.v1_20_R1.CraftServer.enablePlugin(CraftServer.java:563) ~[paper-1.20.jar:git-Paper-5]
> 	at org.bukkit.craftbukkit.v1_20_R1.CraftServer.enablePlugins(CraftServer.java:474) ~[paper-1.20.jar:git-Paper-5]
> 	at org.bukkit.craftbukkit.v1_20_R1.CraftServer.reload(CraftServer.java:1039) ~[paper-1.20.jar:git-Paper-5]
> 	at org.bukkit.Bukkit.reload(Bukkit.java:952) ~[paper-api-1.20-R0.1-SNAPSHOT.jar:?]
> 	at org.bukkit.command.defaults.ReloadCommand.execute(ReloadCommand.java:54) ~[paper-api-1.20-R0.1-SNAPSHOT.jar:?]
> 	at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:155) ~[paper-api-1.20-R0.1-SNAPSHOT.jar:?]
> 	at org.bukkit.craftbukkit.v1_20_R1.CraftServer.dispatchCommand(CraftServer.java:929) ~[paper-1.20.jar:git-Paper-5]
> 	at org.bukkit.craftbukkit.v1_20_R1.CraftServer.dispatchServerCommand(CraftServer.java:892) ~[paper-1.20.jar:git-Paper-5]
> 	at net.minecraft.server.dedicated.DedicatedServer.handleConsoleInputs(DedicatedServer.java:501) ~[paper-1.20.jar:git-Paper-5]
> 	at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:448) ~[paper-1.20.jar:git-Paper-5]
> 	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1396) ~[paper-1.20.jar:git-Paper-5]
> 	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1173) ~[paper-1.20.jar:git-Paper-5]
> 	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:320) ~[paper-1.20.jar:git-Paper-5]
> 	at java.lang.Thread.run(Thread.java:1589) ~[?:?]
> ```
> 
> I believe Bukkit updated SnakeYaml recently

Yea, i removed  SnakeYaml from build and compiled, works well now

**Comment by rudde0** [comment]
> Yea, i removed SnakeYaml from build and compiled, works well now

Could you share your build?

**Comment by memosoff** [comment]
> Could you share your build?

[download link](https://dropmefiles.com/pbuLK)

I really don't know for what SnakeYaml was used, but i removed it and all work perfectly xd

**Comment by DoogiePC** [comment]
> > Could you share your build?
> 
> [download link](https://dropmefiles.com/pbuLK)
> 
> I really don't know for what SnakeYaml was used, but i removed it and all work perfectly xd

It has a password on it...

**Comment by ShaneBeee** [comment]
rather than using some build that stripped out part of the plugin... here is a build that is actually fixed:
(I forked it and fixed the issues on 1.20)
https://github.com/ShaneBeee/skript-yaml/releases
